server/src/main/java/io/crate/analyze/AlterTableDropColumnAnalyzer.java

```diff
@@ -120,5 +140,7 @@ private static void validate(DocTableInfo tableInfo, AnalyzedTableElements<Symbo
         if (leftOverCols.isEmpty()) {
             throw new UnsupportedOperationException("Dropping all columns of a table is not allowed");
         }
+
+        return updatedDropColDefs;
```

Comment:
Looks like we can end up with an empty list if all items don't exists and has "if exists".

We cannot achieve empty list with regular SQL statement but I guess here it's fine as it's about skipping if exists. 
Maybe can add a comment that if list is empty, planner will short-circuit to immediate result.